bridges/otelzap/encoder.go

```diff
@@ -6,13 +6,47 @@ package otelzap // import "go.opentelemetry.io/contrib/bridges/otelzap"
 import (
 	"fmt"
 	"reflect"
+	"sync"
 	"time"
 
 	"go.uber.org/zap/zapcore"
 
 	"go.opentelemetry.io/otel/log"
 )
 
+var arrayEncoderPool = sync.Pool{
+	New: func() interface{} {
+		// Similar to console_encoder which uses capacity of 2
+		return &arrayEncoder{elems: make([]log.Value, 0, 2)}
+	},
+}
+
+func getArrayEncoder() (arr *arrayEncoder, free func()) {
+	arr = arrayEncoderPool.Get().(*arrayEncoder)
+	return arr, func() {
+		arr.elems = arr.elems[:0]
+		arrayEncoderPool.Put(arr)
+	}
+}
+
+var objectEncoderPool = sync.Pool{
+	New: func() interface{} {
+		// Similar to console_encoder which uses capacity of 2
+		return newObjectEncoder(2)
+	},
+}
+
+func getObjectEncoder() (obj *objectEncoder, free func()) {
+	obj = objectEncoderPool.Get().(*objectEncoder)
+	return obj, func() {
+		obj.root.attrs = obj.root.attrs[:0]
```

Comment:
Can you add this comment?

https://github.com/open-telemetry/opentelemetry-go-contrib/blob/c80e46445e3e1726dda9e9fc0c34e255f2ff908e/bridges/otelslog/handler.go#L352-L354